components/TweetModal.vue

```diff
@@ -1,46 +1,108 @@
 <template>
-			<b-modal id='TwitterNftShareModal' title='Congratulations 🎉' centered @hidden="$emit('hidden')">
-         <div class="text-center">
-        You've successfully minted {{mintCount}} NFTs from {{$siteConfig.smartContract.name}} NFT collection.
-      </div>
-      <div class="row d-flex justify-content-center">
-      <div class="col-6 my-2" v-for="(data,i) in images" :key="i">
-        <h5>{{data.name}}</h5>
-        <b-img-lazy width="200px" blank-color="black" blank-width="200px"  :src="data.imageSrc" />
-      </div>
-      </div>
-     
-<template slot="modal-footer">
-  <ShareNetwork
-              ref="twitterShare"
-              network="twitter"
-              :url="$siteConfig.url + '\n'"
-              :title="compTitle"
-              hashtags="zerocodenft,nocode,nft,mint"
-            >
-              <b-icon icon="twitter"></b-icon> Tweet
-            </ShareNetwork>
-  </template>
-    </b-modal>
+	<b-modal
+		id="TwitterNftShareModal"
+		title="Congratulations 🎉"
+		centered
+		@hidden="$emit('hidden')"
+		hide-footer
+	>
+		<div class="text-center">
+			You've successfully minted {{ mintCount }} NFT{{
+				mintCount > 1 ? 's' : ''
+			}}
+			from {{ $siteConfig.smartContract.name }} NFT collection.
+			<br />
+			<ShareNetwork
+				ref="twitterShare"
+				network="twitter"
+				:url="marketplaceURL + '\n'"
+				:title="compTitle"
+				:hashtags="compHashTags.toString()"
+			>
+				<b-img src="@/assets/img/logos/twitter.svg" width="32px" />
+				<span>
+					Share on Twitter!
+				</span>
+			</ShareNetwork>
+		</div>
+		<div class="row d-flex justify-content-center">
+			<div class="col-6 my-2" v-for="(data, i) in images" :key="i">
+				<h6 class="text-muted text-center">{{ data.name }}</h6>
+				<b-img-lazy
+					class="rounded border border-success"
+					width="200px"
+					blank-color="black"
+					blank-width="200px"
+					:src="data.imageSrc"
+				/>
+			</div>
+		</div>
+	</b-modal>
 </template>
 
-<script >
+<script>
 export default {
-  props: {
-    images: Array,
-    mintCount: {
-      type: Number,
-      default:1
-    }
-  },
-  computed: {
-    compTitle() {
-      return `I've just minted ${this.mintCount} NFTs from ${this.$siteConfig.smartContract.name} NFT Collection.\n Mint one for yourself.`
-    }
-  },
+	props: {
+		images: Array,
+		mintCount: {
+			type: Number,
+			default: 1,
+		},
+	},
+	data() {
+		return {
+			marketplaceURL: this.$siteConfig.marketplaceURL,
+		}
+	},
+	async created() {
+		if (!this.marketplaceURL) {
+			try {
+				const isSupportedByOS = [5, 1, 80001, 137, 1001, 8217].includes(
```

Comment:
this could also be a separate util function. also maybe we should create an enum to label these chain IDs and use that instead so it doesn't look like [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))